llvm/test/Feature/Repo/repo_GOIMapCrash.ll

```diff
@@ -0,0 +1,53 @@
+; RUN: opt < %s -S -o %t
+; Test for a bug in  'RepoMetadataGenerationPass' which caused a crash.
```

Comment:
I’m bothered by this test because it doesn’t actually test for anything! Could we check that the dependency/contribution output is as expected?

Presumably we ned to have 47 values because this is point at which (with the current defaults) the underlying buffer is reallocated. Is that correct?

Reply:
(FTR, for me (running on macOS) this test does not fail with the original code unless I compile with AddressSanitizer enabled.)

Reply:
> 
> 
> I’m bothered by this test because it doesn’t actually test for anything! Could we check that the dependency/contribution output is as expected?
> 
Initially, I added this test to check the test crash or not. For the original compiler code, when running the opt command, it crashed and failed. After the fix, it passed the opt command and test passed. 

>
> Presumably we need to have 47 values because this is point at which (with the current defaults) the underlying buffer is reallocated. Is that correct?
>

Yes, it is correct for repo compiler which is built on the windows.

>
>FTR, for me (running on macOS) this test does not fail with the original code unless I compile with AddressSanitizer enabled.
>
After read this comment, I tried to run the test on Linux. It doesn't fail with the original code.
Now, I can't find a way to test the crash on all targets.

@paulhuggett , any suggestion?

Reply:
>
>Yes, it is correct for repo compiler which is built on the windows.
>

The test failed when using the repo compiler with Debug configuration on Windows. However, the test passed when using the repo compiler with Release configuration on Windows.

Reply:
> any suggestion?

It’s not really possible to guarantee that every host platform will crash in the face of undefined behaviour such as writing through dangling pointers. That’s why we have tools like ASan and Valgrind!

It seems likely that the original code won’t produce the expected graph: can you simply check for correct output? You may also be to add an assertion that the pointer is valid using DenseMap<>::isPointerIntoBucketsArray() that would at least provide consistent behaviour on all the hosts.

Reply:
Updated the test and added the assert in the commit https://github.com/SNSystems/llvm-project-prepo/pull/131/commits/78a6f20b42d16c3a08612c798c913e8d25db0a1b.

Still check 47 global variables' dependencies since the test crashed before the fix on Windows.

Reply:
Can you confirm that (prior to the fix) the test now fails on platforms other than Windows?

Reply:
Sorry, It took a long time to build the debug version repo compiler. Now, I can confirm the test failed before the fix and passed after the fix.